### PR TITLE
[Fix] Fix `keypoint_info` of RHD dataset 

### PR DESCRIPTION
**Pull request by liqikai9**
## Motivation

Fix issue #1261.

## Modification

Fix the `skeleton_info` of RHD dataset info under  `mmpose/configs/_base_/datasets/rhd2d.py` by these modifications:

1.`link=('wrist', 'thumb1') `->` link=('wrist', 'thumb4')`

2.`link=('wrist', 'forefinger1')` -> `link=('wrist', 'forefinger4')`

3.l`ink=('wrist', 'middle_finger1') `-> `link=('wrist', 'middle_finger4')`

4.`link=('wrist', 'ring_finger1') `-> `link=('wrist', 'ring_finger4')`

5.`link=('wrist', 'pinky_finger1')` -> `link=('wrist', 'pinky_finger4')`

These modifications are minor, which will not affect the trained models.

Further, fix the paper info of COCO-Wholebody-Hand dataset under `mmpose/docs/en/tasks/2d_hand_keypoint.md`

## Checklist

**Before PR**:

- [ ] I have read and followed the workflow indicated in the [CONTRIBUTING.md](https://github.com/open-mmlab/mmpose/blob/master/.github/CONTRIBUTING.md) to create this PR.
- [ ] Pre-commit or linting tools indicated in [CONTRIBUTING.md](https://github.com/open-mmlab/mmpose/blob/master/.github/CONTRIBUTING.md) are used to fix the potential lint issues.
- [ ] Bug fixes are covered by unit tests, the case that causes the bug should be added in the unit tests.
- [ ] New functionalities are covered by complete unit tests. If not, please add more unit tests to ensure correctness.
- [ ] The documentation has been modified accordingly, including docstring or example tutorials.

**After PR**:

- [ ] CLA has been signed and all committers have signed the CLA in this PR.

### CODE REVIEW

**Comment by ly015** [comment]
@liqikai9 Could you please double-check the keypoint order by visualizing them with the demo script?

**Comment by jin-s13** [comment]
This fix will cause some undesirable confusion. Since in other hand datasets, 'wrist' is connected to 'finger1' (this keypoint has its unique semantic meaning). 

Maybe we should take a look at the definition of RHD hand.

**Comment by liqikai9** [comment]
> @liqikai9 Could you please double-check the keypoint order by visualizing them with the demo script?

Yes, I have checked the demo script using these commands:

`python demo/top_down_img_demo.py configs/hand/2d_kpt_sview_rgb_img/topdown_heatmap/rhd2d/res50_rhd2d_256x256.py checkpoints/res50_rhd2d_256x256-5dc7e4cc_20210330.pth --img-root tests/data/rhd/ --json-file tests/data/rhd/test_rhd.json --out-img-root vis_results_th_rhd`

and 

`python demo/top_down_img_demo.py configs/hand/2d_kpt_sview_rgb_img/deeppose/rhd2d/res50_rhd2d_256x256.py checkpoints/deeppose_res50_rhd2d_256x256-37f1c4d3_20210330.pth --img-root tests/data/rhd/ --json-file tests/data/rhd/test_rhd.json --out-img-root vis_results_dp_rhd`.

And here are the visualization results, respectively.

topdown_heatmap_rhd_vis_0   deeppose_rhd_vis_0

The visualization results look just fine.

**Comment by jin-s13** [comment]
Many users would like to combine different hand datasets to train a single model. Since the keypoint definition is different, it will cause some problems.

**Comment by liqikai9** [comment]
> 

@jin-s13 Yes, itis a little bit confused. Here is the definition of keypoint indices of [RHD dataset](https://lmb.informatik.uni-freiburg.de/resources/datasets/RenderedHandpose/README). 
It says, `1-4: left thumb [tip to palm]`, in which the finger is from tip to palm, so I think the `wrist` should be connected be `finger4`. 

Maybe it is different from the other hand datasets like freihand and coco_wholebody_hand.

**Comment by liqikai9** [comment]
> Many users would like to combine different hand datasets to train a single model. Since the keypoint definition is different, it will cause some problems.

@jin-s13  That's right. But if we change the keypoint indices of these fingers, we may need to train the models related to RHD dataset again, because the order of final channels corresponding to different keypoint dices is different. 

So, do you think it's a good idea to unify these keypoint definitions and train the models again (or add extra process during inference), or can we just change the skeleton connection?

**Comment by jin-s13** [comment]
I think we should add a **warning notice** in RHD dataset. Its keypoint definition is different from others.
Also it is a good idea to rename the keypoint. 1-4 is confusing.

File: mmpose/datasets/datasets/hand/rhd2d_dataset.py [hunk 1]
```diff
@@ -73,6 +73,11 @@ def __init__(self,
                 'for details.', DeprecationWarning)
             cfg = Config.fromfile('configs/_base_/datasets/rhd2d.py')
             dataset_info = cfg._cfg_dict['dataset_info']
+        warnings.warn(
+            'Please note that the in RHD dataset, its keypoint definition is'
+            'different from others. Check `configs/_base_/datasets/rhd2d.py`'
```
Comment (ly015) [inline mmpose/datasets/datasets/hand/rhd2d_dataset.py:78]: 'others' here is ambiguous. Please list some specific dataset names like coco.

**Review by jin-s13 — APPROVED** [review]

**Pull request merged**